text/0000-lifecycle-parallel-export.md

```diff
@@ -0,0 +1,182 @@
+
+# How it Works
+[how-it-works]: #how-it-works
+
+It will be done using goroutines. Goroutine is a lightweight multi-threading mechanism, which can avoid a lot of extra overhead caused by the multi-threaded parallel operation. This function encapsulates the export process of the app image and the cache image into two goroutines to execute in parallel.
+
+The working principle is shown in the following code (go-like pseudocode). Execute in parallel through two goroutines and wait for all export processes to finish. If parallel export is not enabled, the process is exactly the same as the original serial export.
+
+```go
+func export() {
+    exporter := &lifecycle.Exporter{}
+    // ...
+    
+    var wg sync.WaitGroup
+    
+    wg.Add(1)
+    go func() {
+        defer wg.Done()
+        exporter.Export()
+    }()
+	
+    if !enableParallelExport {
+        wg.Wait()
+    }
+    
+    wg.Add(1)
+    go func() {
+        defer wg.Done()
+        exporter.Cache()
+    }()
+    
+    wg.Wait()
+    
+    // ...
+}
+```
+
+## Examples
+
+For command line use, control this process through the `parallel` flag.
+
+### Export both app image and cache image
+
+By specifying environment variable `CNB_PARALLEL_EXPORT`, or pass a `-parallel` flag, images will be pushed to `cr1.example.com` and `cr2.example.com` simultaneously.
+
+```shell
+> export CNB_PARALLEL_EXPORT=true
+> /cnb/lifecycle/exporter -app cr1.example.com/foo:app -cache-image cr2.example.com/foo:cache
+
+# OR
+
+> /cnb/lifecycle/exporter -app cr1.example.com/foo:app -cache-image cr2.example.com/foo:cache -parallel
+```
+
+### Export app image only or export cache image only
+
+If export one image, the effect of this function is not very obvious.
+
+```shell
+> export CNB_PARALLEL_EXPORT=true
+> /cnb/lifecycle/exporter -app cr1.example.com/foo:app
+
+# OR
+
+> /cnb/lifecycle/exporter -app cr1.example.com/foo:app -parallel
+
+# EQUAL TO
+
+> /cnb/lifecycle/exporter -app cr1.example.com/foo:app
+```
+
+# Migration
+[migration]: #migration
+
+We maybe need to add a new API option for buildpack users, to choose whether this feature should be enabled.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+ - This will lead to resource competition, the app image export time will become longer. 
+
+# Alternatives
+[alternatives]: #alternatives
+
+N/A.
+
+# Prior Art
+[prior-art]: #prior-art
+
+N/A.
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- How to allow users to choose the export method? environment variable? Or a parameter of the creator? 
+- Does it also need to be specified in the pack tool?
+- Should this feature be enabled by default?
+
```

Comment:
From my point of view:
- How to allow users to choose the export method? environment variable? Or a parameter of the creator? 
I like how you are proposing it, with the flag the env variable

- Does it also need to be specified in the pack tool?
Probably, once we get the RFC approved, we can create the issue on `pack` to expose the behavior for end-users

- Should this feature be enabled by default?
I'd say **no**  to avoid unexpected behaviors to people consuming the *lifecycle*.